File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/align.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// Align — posiciona o filho dentro de si.
Widget buildAlign(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Align(
    alignment: alignmentFrom(p['alignment']) ?? Alignment.center,
    widthFactor: parseDouble(p['widthFactor']),
    heightFactor: parseDouble(p['heightFactor']),
    child: r.maybeRender(context, node.child),
  );
}

File: docs/WidgetMill/packages/spec/src/nodes/align.ts
import { z } from "zod";
import { Alignment } from "../complex/alignment";
import { NodeRef } from "../node-ref";

/** Align — posiciona o filho dentro de si (catálogo Flutter). */
export const AlignNode = z.object({
  type: z.literal("align"),
  props: z
    .object({
      alignment: Alignment.optional(),
      widthFactor: z.number().optional(),
      heightFactor: z.number().optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});
